src/main/kotlin/com/sksamuel/avro4k/schema/ClassSchemaFor.kt

```diff
@@ -35,8 +36,13 @@ class ClassSchemaFor(
       }
    }
 
+   companion object {
+      // map of already known schema of classes (useful for managing recursive schemas)
+      private val classSchemaBag = ConcurrentHashMap<String, Schema>()
```

Comment:
Having the classSchemaBag in the companion object would make it impossible to use Avro in a multi threaded environment.

Reply:
👍